client/my-sites/domains/domain-management/dataviews/domains-dataviews.tsx

```diff
@@ -0,0 +1,103 @@
+import { PartialDomainData } from '@automattic/data-stores';
+import { DataViews, filterSortAndPaginate } from '@wordpress/dataviews';
+import clsx from 'clsx';
+import { useTranslate } from 'i18n-calypso';
+import { useEffect, useMemo, useState } from 'react';
+import { navigate } from 'calypso/lib/navigate';
+import { BulkUpdateNotice } from './components/bulk-update-notice';
+import { useActions } from './use-actions';
+import { useDomainsDataViewsContext } from './use-context';
+import { getDomainId } from './use-domains';
+import './style.scss';
+import { useFields } from './use-fields';
+import useQueryParams, {
+	DEFAULT_PER_PAGE,
+	DEFAULT_SORT_FIELD,
+	DEFAULT_SORT_DIRECTION,
+	buildPathWithQueryParams,
+} from './use-query-params';
+import useView, { getFieldsByBreakpoint } from './use-view';
+
+type Props = {
+	domains: PartialDomainData[] | undefined;
+	isLoading: boolean;
+	sidebarMode?: boolean;
+	selectedDomainName?: string;
+	openDomainPane?: ( domain: PartialDomainData ) => void;
+};
+
+export const DomainsDataViews = ( {
+	domains,
+	isLoading,
+	sidebarMode,
+	selectedDomainName,
+	openDomainPane,
+}: Props ) => {
+	const translate = useTranslate();
+	const { isDesktop } = useDomainsDataViewsContext();
+	const queryParams = useQueryParams();
+
+	const { view, setView } = useView( { sidebarMode, isDesktop, queryParams } );
```

Comment:
This hook is just a wrapper for the useState's initializer function. It'd be clear if we do:

```js
const [ view, setView ] = useState( () => initializeState( sidebarMode, isDesktop, queryParams ) );
```

Reply:
Fixed here: https://github.com/Automattic/wp-calypso/pull/97677/commits/13c92a89a69fc16f3d57af80fd527888c6f42248